### DISCUSSION TITLE
Ability to import a depth map and ability to output depth map only in iw3

### DISCUSSION THREAD

**Discussion by kaelsonofkrypto** [nagadomi/nunif#87] [Ideas]
Hi if possible can you add an option where you can Allow the option to import your own depth map for iw3 to do the 2D to 3D conversion off your own depth map. If this option is created it would have to work something like 3D combine where you import The 2D video for the left side and then you import the depth map for the right side and then the program will combine the two into an sbs during conversion.

 And also, would it be possible to add the ability to just output a depth map only from converting a video using the depth map models already within the program. The reason I'm asking for this is so that. I could take an IW3 depth map. I created of a video, so I could clean the depth map up in another program, and then once I've cleaned up the depth map and enhanced it, then be able to import the clean version of the depth map back into IW3 to complete the 2D to 3D conversion. Thanks.

**Comment by nagadomi** [comment]
Added support in https://github.com/nagadomi/nunif/issues/97#issuecomment-2027349722

**Reply by kaelsonofkrypto** [reply]
Hi I read the post you mentioned to look for the export in the gui but I don't see the export button this is how it looks for me in the picture below maybe I'm missing something, Thanks for all your help
![dont see export button](https://github.com/nagadomi/nunif/assets/164638883/12c5bb02-6bfb-46ed-8c72-1fa471614c46)

**Reply by nagadomi** [reply]
You can select `Export` with `Stereo Format` option.

**Reply by kaelsonofkrypto** [reply]
thanks, found it

**Comment by kaelsonofkrypto** [comment]
one last question for you, and thank you for everything, you are simply amazing, is there an option where u can import your own depth map into iw3 and iw3 does a conversion off a depth map you imported into it? and if not would this be possible to add one day. thanks again  this is probably the only other thing i would like to see, this is a near perfect program that you have made

**Reply by nagadomi** [reply]
Exported data can be imported by specifying a yaml config file as input.
If you prepare the same format with it, you can import external data.

For video, it requires both rgb(frame) images and depth images, and audio file if needed.
You can use the original video for the audio file if it is in the same range.
If frame images are not available, you can generate it from the original video using ffmpeg or other software. However, it must be in the same order as depth images.

For image, the rgb image is just the original image.

**Reply by kaelsonofkrypto** [reply]
thanks, also checked your post about the difference between regular export and export disparity and that was great information. thanks again You are a pure genius.

**Reply by nagadomi** [reply]
`Export` outputs depth images with only min-max scaling applied. It will output different types of depth depending on the model used (ZoeDepth or DepthAnything). 

`Export disparity` outputs depth images with EdgeFix and transform to disparity(--mapper) applied. Both ZoeDepth and DepthAnything will output disparity.

DepthAnything's output is disparity(relative depth), so both `Export` and `Export Disparity` will output disparity. 
ZoeDepth's output is metric depth, so `Export` and `Export disparity` will output very different result.

**Reply by nagadomi** [reply]
If the depth image is not disparity, it should be transformed to disparity when generating stereo. The transformation method is specified in `mapper` field of the yaml file. If "mapper: none" is specified, the depth image is used as is.
